Updated the rails version in 7.0 to 7.1 in guide [skip ci]

akhilgkrishnan opened the pull request:
### Detail

This Pull Request updates the rails version from 7.0 to 7.1.

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Changes that are unrelated should be opened in separate PRs.
* [ ] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [ ] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

akhilgkrishnan committed:
Updated the rails version in 7.0 to 7.1 in guide [skip ci]

mohits commented:
Are you going to update the rails console outputs also?

 By the way, I feel that saying "if you have something like 7.0.0" is still fine. Instead, updating the Ruby comment to be at least Ruby 3.0, is probably a good idea.

hieuk09 committed:
SQLite - Remove quote when extracting table ref in insert query

zzak commented:
Could you make sure we get all references in a single PR?

See: 35bf079aed8b00ef92bf394e32978cf76b9795d1 [#41894, #42113]

akhilgkrishnan commented:
> Could you make sure we get all references in a single PR?
> 
> See: [35bf079](https://github.com/rails/rails/commit/35bf079aed8b00ef92bf394e32978cf76b9795d1) [#41894, #42113]

@zzak This complete change should go in `7-1-stable`, as the main branch use the 7.2.

zzak commented:
I don't feel strongly about this, but that makes sense.

Thomascountz committed:
Fix `RedisCacheStore` INCR/DECR for Redis < v7.0.0

This commit fixes a discrepancy in the behavior of the `#increment` and
`#decrement` methods in `RedisCacheStore` when used with Redis versions less
than 7.0.0. The existing condition `count != amount` prevented setting the
Time-To-Live (TTL) for keys that were equal to the increment/decrement amount
after the `INCRBY`/`DECRBY` operation. This occurs when incrementing a
non-existent key by `1`, for example.

Using Redis pipelining, we minimize the network overhead incurred by checking
for existing TTLs. It decouples the TTL operations from the increment/decrement
operation, allowing the TTL to be set correctly regardless of the resulting
value from the `INCRBY`/`DECRBY`.

New tests have been added to verify the correct behavior of `#increment` and
`#decrement` methods, specifically when the `expires_in` option is not used.
Using a separate cache store instance (`@cache_no_ttl`), these tests ensure that
keys are correctly incremented or decremented and that their TTL remains unset.

Co-authored-by: Benjamin Quorning <benjamin@quorning.net>
Co-authored-by: Jury Razumau <jury.razumau@zendesk.com>
Co-authored-by: Edyta Rozczypała <edyta.rozczypala@zendesk.com>

akhilgkrishnan committed:
Update postgres naming to PostgreSQL

tricknotes committed:
[Refactor] Remove unused from tests for `@rails/ujs`

jenshenny committed:
Add support for kwargs when delegating calls to custom loggers

Currently, when a method is called on Rails.logger, the BroadcastLogger will find the loggers that will respond to that method. However, when the method has keyword arguments, they are passed as regular arguments and will throw an ArgumentError. This adds keyword argument support by double splatting hash args.

```
class CustomLogger
  def foo(bar:)
    true
  end
end

Rails.logger.foo(bar: "baz")
```

Expected: `true`

Actual: `wrong number of arguments (given 1, expected 0) (ArgumentError)`

jbampton committed:
guides/source/configuring.md: fix grammar and spelling

jbampton committed:
guides/source/asset_pipeline.md: change `Javascript` to `JavaScript`

akhilgkrishnan committed:
Merge branch 'rails:main' into main

akhilgkrishnan commented:
@zzak I've changed the base to `7-1-stable`.
